static/app/components/discover/performanceCardList.tsx

```diff
@@ -0,0 +1,129 @@
+import EventView from 'app/utils/discover/eventView';
+import {decodeScalar} from 'app/utils/queryString';
+import {SpanOperationBreakdownFilter} from 'app/views/performance/transactionSummary/filter';
+import {TransactionFilterOptions} from 'app/views/performance/transactionSummary/utils';
+
+import PerformanceCardTable from './performanceCardTable';
+
+const DEFAULT_TRANSACTION_LIMIT = 5;
+
+type Props = {
+  location: Location;
+  eventView: EventView;
+  releaseEventView: EventView;
+  organization: Organization;
+  project: ReleaseProject;
+  /**
+   * The name of the url parameter that contains the cursor info.
+   */
+  cursorName: string;
+  /**
+   * The limit to the number of results to fetch.
+   */
+  limit: number;
+  /**
+   * A map of callbacks to generate a link for a column based on the title.
+   */
+  generateLink?: Record<
+    string,
+    (
+      organization: Organization,
+      tableRow: TableDataRow,
+      query: Query
+    ) => LocationDescriptor
+  >;
+  /**
+   * Show a loading indicator instead of the table, used for transaction summary p95.
+   */
+  forceLoading?: boolean;
+  /**
+   * Optional callback function to generate an alternative EventView object to be used
+   * for generating the Discover query.
+   */
+  generateDiscoverEventView?: () => EventView;
+  generatePerformanceTransactionEventsView?: () => EventView;
+  showTransactions?: TransactionFilterOptions;
+  breakdown?: SpanOperationBreakdownFilter;
+};
+
+class PerformanceCardList extends React.Component<Props> {
+  static defaultProps = {
+    cursorName: 'transactionCursor',
+    limit: DEFAULT_TRANSACTION_LIMIT,
+  };
+
+  handleCursor = (cursor: string, pathname: string, query: Query) => {
+    const {cursorName} = this.props;
+    browserHistory.push({
+      pathname,
+      query: {...query, [cursorName]: cursor},
+    });
+  };
+
+  renderTransactionTable(): React.ReactNode {
+    const {
+      location,
+      organization,
+      project,
+      cursorName,
+      limit,
+      generateLink,
+      forceLoading,
+      eventView,
+      releaseEventView,
+    } = this.props;
+    const columnOrder = eventView.getColumns();
+    const cursor = decodeScalar(location.query?.[cursorName]);
+
+    const tableRenderer = ({isLoading, tableData, releaseTableData}) => (
+      <React.Fragment>
+        <PerformanceCardTable
+          eventView={eventView}
+          releaseEventView={releaseEventView}
+          organization={organization}
+          project={project}
+          location={location}
+          isLoading={isLoading}
+          tableData={tableData}
+          releaseTableData={releaseTableData}
+          columnOrder={columnOrder}
+          generateLink={generateLink}
+        />
+      </React.Fragment>
+    );
+
+    if (forceLoading) {
+      return tableRenderer({
+        isLoading: true,
+        tableData: null,
+        releaseTableData: null,
+      });
+    }
+
+    return (
+      <DiscoverQuery
```

Comment:
`genericDiscoverQuery` is intended to be generic, bringing in specific implementation detailed from performanceTable isn't the best.

What we usually do is compose two queries together by wrapping one discover query specialization with another. I'd recommend making a specialization (like [this](https://github.com/getsentry/sentry/blob/5eb101673045a343bbec797b69dcd558d5c07152/static/app/utils/performance/histogram/histogramQuery.tsx), or [this](https://github.com/getsentry/sentry/blob/5eb101673045a343bbec797b69dcd558d5c07152/static/app/utils/performance/vitals/vitalsDetailsTableQuery.tsx), etc.), and simply wrap the first discover query with your specialization.

```js
<DiscoverQuery
  ...
> 
 (...discoverData) =>
  <ReleaseTableQuery>
     (...releaseData) => {
        // do composition here.
     }
  </ReleaseTableQuery>
</DiscoverQuery>
```

That way all the code for composition can occur solely inside your component instead of leaking into other performance query implementations.

If the generic discover query doesn't work in some way for your use case, let me know. Any modification to it should be agnostic to the implementation, so we'd look into the functionality missing to enable you to compose externally.